examples/ann/main.go

```diff
@@ -0,0 +1,238 @@
+		bit := tempInput & 1
+		tempInput = tempInput >> 1
+		if bit == 0 {
+			binary += "0"
+		} else {
+			binary += "1"
+		}
+		if _, err := tx.Exec(`UPDATE nodes SET value=$1 WHERE id=$2`, float64(bit), getNodeID(0, i)); err != nil {
+			log.Fatal(err)
+		}
+		// TODO(bram): Obviously, remove this once we have feed forward working.
+		if _, err := tx.Exec(`UPDATE nodes SET value=$1 WHERE id=$2`, float64(bit), getNodeID(totalLayers-1, i)); err != nil {
+			log.Fatal(err)
+		}
+	}
+	if err = tx.Commit(); err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("Input: %d or %s\n", newInput, binary)
+	return newInput
+}
+
+func getOutput() int64 {
+	var result int64
+	var binary string
+	rows, err := db.Query(`
+SELECT id, value
+FROM nodes
+WHERE id >= $1
+AND id < $2
+ORDER BY id DESC
+		`, getNodeID(totalLayers-1, 0), getNodeID(totalLayers-1, *outputLayerNodes))
+	if err != nil {
+		log.Fatal(err)
+	}
+	var length int
+	for rows.Next() {
+		length++
+		var id int64
+		var value float64
+		if err := rows.Scan(&id, &value); err != nil {
+			log.Fatal(err)
+		}
+		bit := !(value < 0.5)
+		result = result << 1
+		if bit {
+			result = result | 1
+			binary = "1" + binary
+		} else {
+			binary = "0" + binary
+		}
+	}
+	if length != *outputLayerNodes {
+		log.Fatalf("Incorrect output nodes in query. Expected %d, got %d.\n", *outputLayerNodes, length)
+	}
+	fmt.Printf("Output: %d or %s\n", result, binary)
+	return result
+}
+
+func main() {
+	rng = rand.New(rand.NewSource(time.Now().UnixNano()))
+	flag.Parse()
+	var err error
+	var url string
+	security.SetReadFileFn(securitytest.Asset)
+	serv := server.StartTestServer(nil)
+	defer serv.Stop()
+	url = "https://root@" + serv.ServingAddr() + "?certs=test_certs"
+	if db, err = sql.Open("cockroach", url); err != nil {
+		log.Fatal(err)
+	}
+
+	if *inputLayerNodes <= *outputLayerNodes {
+		maxInput = int64(math.Sqrt(math.Exp2(float64(*inputLayerNodes))))
+	} else {
+		maxInput = int64(math.Sqrt(math.Exp2(float64(*outputLayerNodes))))
+	}
+
+	var widestLayer int
+	if *inputLayerNodes >= *hiddenLayerNodes && *inputLayerNodes >= *outputLayerNodes {
+		widestLayer = *inputLayerNodes
+	} else if *hiddenLayerNodes >= *inputLayerNodes && *hiddenLayerNodes >= *outputLayerNodes {
+		widestLayer = *hiddenLayerNodes
+	} else {
+		widestLayer = *outputLayerNodes
+	}
+	db.SetMaxOpenConns(widestLayer + 1)
+	totalLayers = 2 + *hiddenLayers
+
+	fmt.Printf("Artificial Neural Network Simulator\n")
+	fmt.Printf("Input Nodes: %d\n", *inputLayerNodes)
+	fmt.Printf("Hidden Layers: %d\n", *hiddenLayers)
+	fmt.Printf("Hidden Nodes per Layer: %d\n", *hiddenLayerNodes)
+	fmt.Printf("Output Nodes: %d\n", *outputLayerNodes)
+	fmt.Printf("Total Layers: %d\n", totalLayers)
+	fmt.Printf("Max Input: %d\n", maxInput)
+	fmt.Printf("Concurrent Connections: %d\n", widestLayer+1)
+	fmt.Printf("Total Iterations: %d\n", *totalIterations)
+	// Create the database.
+	if _, err := db.Exec("CREATE DATABASE ann"); err != nil {
```

Comment:
since you are not dropping this database anywhere, you should use `CREATE DATABASE IF NOT EXISTS` here